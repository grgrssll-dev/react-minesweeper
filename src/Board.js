import React, { Component } from 'react';
import './Board.css';
import Cell from './Cell';

class Board extends Component {

	constructor(props) {
		super(props);
		this.state = {
			clicks: 0
		};
		this._onCellClickBound = this._onCellClick.bind(this);
	}

	_onCellClick(isRightClick, x, y) {
		const { onCellClick, onMineFlag } = this.props;
		if (isRightClick) {
			onMineFlag(x, y);
		} else {
			this.setState((state) => {
				return {
					clicks: state.clicks + 1
				};
			}, () => {
				onCellClick(x, y);
			});
		}
	}

	render() {
		const { rows, cols, data } = this.props;
		console.log('-- board render', rows, cols);

		return (
			<div className="board">
				{data.map((row, x) => {
					return row.map((cell, y) => {
						return (
							<Cell key={`${x},${y}`}
								triggered={cell.triggered}
								x={cell.x}
								y={cell.y}
								number={cell.number}
								isFlagged={cell.isFlagged}
								isRevealed={cell.isRevealed}
								onCellClick={this._onCellClickBound} />
						);
					}).concat((<br key={`br-${x}`} />));
				})}
			</div>
		);
	}
}

export default Board;
