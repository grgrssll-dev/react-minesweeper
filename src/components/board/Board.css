.board {
	border-bottom-color: var(--game-border-light);
	border-left-color: var(--game-border-dark);
	border-right-color: var(--game-border-light);
	border-style: solid;
	border-top-color: var(--game-border-dark);
	border-width: 0.125rem;
	font-size: 0;
}

.board br {
	width: 0;
	height: 0;
	display: block;
}