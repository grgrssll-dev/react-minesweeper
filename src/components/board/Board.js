import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import Cell from '../cell/Cell';
import { cellDataType } from '../../propTypes';

function board(props) {
	const {
		rows,
		cols,
		data,
		onCellClick,
		onMineFlag,
		isGameOver,
		className,
	} = props;

	const onClick = (isRightClick, x, y) => {
		if (!isGameOver) {
			if (isRightClick) {
				onMineFlag(x, y);
			} else {
				onCellClick(x, y);
			}
		}
	}

	console.log('-- board render', rows, cols);
	/* eslint-disable react/no-array-index-key */
	return (
		<div className={`${className} board`}>
			{data.map((row, x) => {
				return row.map((cell, y) => {
					return (
						<Cell key={`${x},${y}`}
							triggered={cell.triggered}
							x={cell.x}
							y={cell.y}
							isGameOver={isGameOver}
							number={cell.number}
							isFlagged={cell.isFlagged}
							isRevealed={cell.isRevealed}
							onCellClick={onClick} />
					);
				}).concat((<br key={`br-${x}`} />));
			})}
		</div>
	);
	/* eslint-ensable react/no-array-index-key */
}

board.propTypes = {
	rows: PropTypes.number.isRequired,
	cols: PropTypes.number.isRequired,
	data: PropTypes.arrayOf(PropTypes.arrayOf(cellDataType)).isRequired,
	onCellClick: PropTypes.func.isRequired,
	onMineFlag: PropTypes.func.isRequired,
	isGameOver: PropTypes.bool.isRequired,
	className: PropTypes.string.isRequired,
}

export default styled(board)`
	border-bottom-color: var(--game-border-light);
	border-left-color: var(--game-border-dark);
	border-right-color: var(--game-border-light);
	border-style: solid;
	border-top-color: var(--game-border-dark);
	border-width: 0.125rem;
	font-size: 0;

	& br {
		width: 0;
		height: 0;
		display: block;
	}
`;
