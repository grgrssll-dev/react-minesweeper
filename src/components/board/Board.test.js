import React from 'react';
import { shallow } from 'enzyme/build';
import Board from './Board';

const baseProps = {
	rows: 10,
	cols: 10,
	data: [[]],
	onCellClick: jest.fn(),
	onMineFlag: jest.fn(),
	isGameOver: false,
	className: 'class-name',
};

describe('Board [Unit]', () => {
	beforeEach(() => {
		jest.clearAllMocks();
	});

	it('should match component snapshot', () => {
		expect.assertions(1);
		const props = {
			...baseProps,
		};
		const wrapper = shallow(<Board {...props} />);
		expect(wrapper).toMatchSnapshot();
	});
});