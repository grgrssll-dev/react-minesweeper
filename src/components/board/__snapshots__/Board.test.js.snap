// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board [Unit] should match component snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
